livekit-plugins/livekit-plugins-openai/livekit/plugins/openai/realtime/realtime_model.py

```diff
@@ -660,6 +685,12 @@ async def aclose(self) -> None:
     def chat_ctx(self) -> llm.ChatContext:
         return self._chat_ctx
 
+    @chat_ctx.setter
+    def chat_ctx(self, chat_ctx: llm.ChatContext) -> None:
+        """Sync the ctx to the session and reset the chat context."""
+        self._sync_chat_ctx_to_session(self._chat_ctx, chat_ctx)
+        self._chat_ctx = chat_ctx
```

Comment:
I don't think we should add a setter. Instead we should have a public `sync_ctx` function.
I guess RealtimeModel will needs to have 2 llm.ChatContext (one with the current server states, and one that the user can edit)

Reply:
Yeah, I am thinking something similar that we should keep a "copy" of server state in the RealtimeModel that user cannot edit. I'll add that.

Reply:
Created a `_conversation_items` (maybe give it another name later) in the realtime session to track the item created and deleted events from the API. The content of the item like user and agent transcriptions and the function call outputs will be updated to the item from the multimodal agent.

The `_session.chat_ctx` will be created from the tracked conversation items and user can sync a `chat_ctx` to the session as well.